### ISSUE TITLE
fsave/frstor/fstenv/fldenv instructions do not recognize operand size override prefix (66h)

### ISSUE THREAD

**Issue by pcdosretro** [joncampbell123/dosbox-x#3440]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar bug report(s) already exists?

- [X] I have searched and didn't find any similar issues.

### What operating system(s) this bug have occurred on?

Windows 7 SP 1 64-bit

### What version(s) of DOSBox-X have this bug?

0.83.24

### Describe the bug

fsave/frstor/fstenv/fldenv instructions do not recognize operand size override prefix (66h)

### Expected behavior

For example fnsave ds:[80h] ; opcodes 66 DD 36 80 00 should use 32-bit format and store FPU registers at offset 28 (not 14)

### Steps to reproduce the behaviour

DEBUGX from my website uses this in the RN and RNR commands to display FPU registers
https://pcdosretro.github.io/enhdebug.htm

Within DEBUGX enter:
f 80 l80 0
a 100
fninit
fld1
fld1
fldpi
fldpi
db 66
fnsave [80]
int 3

g
rn
d 80

---
rn should show the correct values (1 twice and pi twice) in the FPU registers but doesn't because the operand size prefix isn't handled on fnsave. rnr is the same but displays the registers in raw format.

**Comment by pcdosretro** [comment]
I believe something like this may fix the problem by having a 66h prefix toggle the size usage:

In the files:
fpu_instructions.h
fpu_instructions_longdouble.h
fpu_instructions_x86.h

In functions FPU_FSTENV and FPU_FLDENV change:
if(!cpu.code.big) {
to:
if (!cpu.code.big ^ (last_prefix == MP_66)) {

In functions FPU_FSAVE and FPU_FRSTOR change:
Bitu start = (cpu.code.big?28:14);
to:
Bitu start = (cpu.code.big ^ (last_prefix == MP_66)) ? 28:14;

Note: fpu_instructions.h uses uint8_t instead of Bitu

**Comment by cimarronm** [comment]
@pcdosretro Good catch for the FPU accuracy with this and the `fldxxx` instructions. PR #3485 should address this one for all the emulations cores.
